zipline/finance/blotter.py

```diff
@@ -162,7 +162,8 @@ def execute_cancel_policy(self, event):
             warn = self.cancel_policy.warn_on_cancel
 
             for orders_by_sid in itervalues(self.open_orders):
-                for order in orders_by_sid:
+                # So we don't mutate the list as we cancel orders
+                for order in orders_by_sid[:]:
```

Comment:
Doesn't directly fit because of the warning messages, but if you think it makes more sense, I could refactor the code to make it use that method